### ISSUE TITLE
not support notion database

### ISSUE THREAD

**Issue by pkwenda** [souvikinator/notion-to-md#22]
I wanna convert the database to markdown [table] but is an empty result.

![image](https://user-images.githubusercontent.com/14212375/155696051-8c2f4cb0-6188-4090-8e71-92f2b70405b0.png)

![image](https://user-images.githubusercontent.com/14212375/155696186-3e8447ee-f0ff-4184-8c74-4e89c6111a55.png)

can you fix it? 🤓

**Comment by souvikinator** [comment]
yes at the moment database to markdown is not supported as a database can have multiple views. I can view the same database table as a list, board, gallery, and so on.

If you want to use tables, you can go with notion's **[simple table](https://www.notion.so/help/guides/simple-tables-vs-databases)**  which is supported by notion-to-md.

**Comment by pkwenda** [comment]
@souvikinator Ok, I understand, I am looking for a notion database to MD method. Because many people don't use a simple table， thank you，Thank you for your reply ！

**Comment by souvikinator** [comment]
@kryp2z

**Comment by kmlbgn** [comment]
@souvikinator 
> yes at the moment database to markdown is not supported as a database can have multiple views. I can view the same database table as a list, board, gallery, and so on.
> 
> If your requirement is to use tables, you can go with notion's **[simple table](https://www.notion.so/help/guides/simple-tables-vs-databases)** which is supported by notion-to-md.



Would it be possible to support database in table view in that sense ?

**Comment by souvikinator** [comment]
Hi! Thank you for raising this issue. 

We're currently working on **notion-to-md v4** which includes significant architectural improvements so this can be handled along the way.

I'd love your insights on this. Please checkout the GitHub discussion #112   to discuss how we can best solve your use case? Your feedback would be really valuable in shaping v4's development.

**Comment by oliverkaiser** [comment]
Just came across this and think is can be very valuable. 

As usecase, e.g. we are building an extensive internal knowledgebase and have then database views embedded on quite some pages to represent the latest information. Typically we have one view, but I can see the reason for multiple views.

From a user perspective I see it this way:
- every content piece that is visible on a page on notion, should be exported
- that means: the data presented in embedded databases or views, which can also be only a filtered set of data. The excluded data clearly shouldn't appear in the export. 
- if there are multiple views, this could lead to the export of multiple tables after each other, or only the first view as default.

**Comment by DeV1L** [comment]
Hi @souvikinator ,
Does `v4` solve the issue with a database object type somehow?

**Comment by souvikinator** [comment]
@DeV1L v4 **does** support `child_databases`, but it requires some configuration. This is a key priority for the upcoming `alpha.5` release.

@oliverkaiser, it's important to note that the Notion API does **not** provide direct access to retrieve or manage database views. It only returns database-related content and metadata.  

That said, if you're using **v4 alpha**, you can leverage **custom transformers** to format and display the data in any way you need.  

**Example: Custom Transformer for Child Databases**  
Here's a short example demonstrating how you can fetch a child database and transform it for display:

```typescript
renderer.createBlockTransformer("child_database", {
  transform: async ({ block, manifest }) => {
    // Fetch database metadata
    const databaseInfo = await notionClient.databases.retrieve({
      database_id: block.id,
    });
    console.log("Database Info:", JSON.stringify(databaseInfo, null, 2));

    // Fetch database content
    const response = await notionClient.databases.query({
      database_id: block.id,
    });
    console.log(JSON.stringify(response.results, null, 2));

    // Define custom logic for processing and rendering the database
    const databaseView: string = convertChildDatabaseToView(databaseInfo, response.results);

    return databaseView;
  },
});
```

The upcoming `alpha.5` release will introduce:  
1. **Automatic Fetching:** By default, it will retrieve both database metadata and content, eliminating the need to fetch it manually within the transformer.  
2. **Default Markdown Rendering:** If no custom transformer is provided, the database will default to a simple Markdown table, as it aligns best with Markdown's capabilities.

For more customized needs, you have two options:  
1. **Custom Transformers:** Define your own logic to process and display the database block in the desired format (above example minus the explicit fetching of database content).  
2. **Custom Renderer Plugin:** If Markdown/MDX doesn't fit your requirements, you can [create a dedicated renderer plugin](https://docs.joinescape.org/notion-to-md/docs/v4/guides/how-to-create-renderer-from-scratch/) for more advanced rendering.  


Lemme know your thoughts.

**Comment by DeV1L** [comment]
@souvikinator ,
> If no custom transformer is provided, the database will default to a simple Markdown table

This is exactly what's needed. In my case, the databases objects are just pages with tables. I was surprised that these aren't actually tables under the hood :) 

Could it be done with v4 and custom transformers?

**Comment by oliverkaiser** [comment]
@souvikinator great thanks.

But just to understand it write, as you say views are not supported, will the export contain then the full DB or if its just a filtered DB, only the filtered rows?

**Comment by souvikinator** [comment]
@oliverkaiser That depends on the Notion API call and the query being passed.  

By default, I was considering fetching the entire database (i.e., making a Notion API call without applying any filters or queries). If needed, we can apply custom transformations on the retrieved data (get's complicated if page uses multiple databases of different structures and kinds). However, if you have a better approach in mind, feel free to share.  

Another option is to let users create mappings for database IDs against specific filters. Users could define filters (similar to those used in Notion API calls), which would then be applied when fetching data.  

Let me know your thoughts.

**Comment by souvikinator** [comment]
@oliverkaiser @DeV1L  
The work is nearly complete! It took more effort than expected, but it's done, and highly customizable and extensible. Now wrapping up the docs to make everything clear and ensure a smooth experience using the package. 🚀
Will share further the updates here.

**Comment by klokie** [comment]
Hi! Would you mind commenting on the status of db => md conversion? Thanks!
